include/boost/multiprecision/cpp_modular.hpp

```diff
@@ -0,0 +1,84 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2019 Nil Foundation AG
+// Copyright (c) 2019 Mikhail Komarov <nemo@nilfoundation.org>
+// Copyright (c) 2019 Alexey Moskvin
+//
+// Distributed under the Boost Software License, Version 1.0
+// See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt
+//---------------------------------------------------------------------------//
+
+#ifndef BOOST_MP_CPP_MODULAR_HPP
+#define BOOST_MP_CPP_MODULAR_HPP
+
+#include <boost/multiprecision/modular/modular_params.hpp>
+#include <boost/multiprecision/modular/modular_adaptor.hpp>
+#include <boost/multiprecision/cpp_int.hpp>
+
+namespace boost {
+namespace multiprecision {
+
+//mnt params
+typedef modular_params<cpp_int_backend<> > cpp_mod_params;
+
+// Fixed precision unsigned types:
+typedef modular_params<cpp_int_backend<128, 128, unsigned_magnitude, unchecked, void> >   umod_params_params128_t;
```

Comment:
Just a quick question - are fixed precision types appropriate here?  Which is to say do they still work correctly if the operation prior to the modular reduction would overflow?

Reply:
That should work fine because modular params depend on internal limb storage size, which is non-mutable in case of fixed precision type. So, if the number was initialized properly with particular number, it should work fine.